event_sourcery_sqlalchemy/__init__.py

```diff
@@ -35,27 +53,46 @@ class Config(BaseModel):
 
 
 class SQLAlchemyBackend(TransactionalBackend):
+    UNCONFIGURED_MESSAGE = "Configure backend with `.configure(session, config)`"
+
     def __init__(self) -> None:
         super().__init__()
-        self[Session] = not_configured(
-            "Configure backend with `.configure(session, config)`",
-        )
-        self[Config] = not_configured(
-            "Configure backend with `.configure(session, config)`",
-        )
+        self[BaseEvent] = not_configured(self.UNCONFIGURED_MESSAGE)
+        self[BaseStream] = not_configured(self.UNCONFIGURED_MESSAGE)
+        self[BaseSnapshot] = not_configured(self.UNCONFIGURED_MESSAGE)
+        self[BaseOutboxEntry] = not_configured(self.UNCONFIGURED_MESSAGE)
+        self[Session] = not_configured(self.UNCONFIGURED_MESSAGE)
+        self[Config] = not_configured(self.UNCONFIGURED_MESSAGE)
         self[StorageStrategy] = lambda c: SqlAlchemyStorageStrategy(
             c[Session],
             c[Dispatcher],
             c.get(SqlAlchemyOutboxStorageStrategy, None),
+            c[BaseEvent],
+            c[BaseSnapshot],
+            c[BaseStream],
         ).scoped_for_tenant(c.tenant_id)
         self[SubscriptionStrategy] = lambda c: SqlAlchemySubscriptionStrategy(
             c[Session],
             c[Config].gap_retry_interval,
+            c[BaseEvent],
+            c[BaseStream],
         )
 
-    def configure(self, session: Session, config: Config | None = None) -> Self:
+    def configure(
+        self,
+        session: Session,
+        config: Config | None = None,
+        event_model: type[BaseEvent] = DefaultEvent,
+        stream_model: type[BaseStream] = DefaultStream,
+        snapshot_model: type[BaseSnapshot] = DefaultSnapshot,
+        outbox_entry_model: type[BaseOutboxEntry] = DefaultOutboxEntry,
+    ) -> Self:
         self[Session] = session
         self[Config] = config or Config()
+        self[BaseEvent] = lambda _: event_model
```

Comment:
It should work with `self[BaseEvent] = event_model`

Reply:
It doesn't work, it treats `event_model` as provider and tries to call it